src/utils/time-entry-utils/combineTimeErrors.jsx

```diff
@@ -1,47 +1,94 @@
-import TimeInputErrors from '../../components/timecard/time-input-errors/TimeInputErrors';
+import { createClashErrorsObject } from '../../components/timecard/time-input-errors/TimeInputErrors';
 import { clashingProperties, inputNames } from '../constants';
 
 export function combineExistingAndTimeClashErrors(
-  errors,
   summaryErrors,
   clashingProperty,
-  clashingTimes
+  clashingTimes,
+  timePeriodTypesMap
 ) {
-  const existingErrors =
-    Object.keys(errors).length > 0 ? errors : summaryErrors;
-
-  const combinedErrors = { ...existingErrors };
+  let combinedErrors = [...summaryErrors];
 
   if (clashingProperty === clashingProperties.startAndEndTime) {
-    combinedErrors[inputNames.shiftStartTime] = clashErrorMessage(
-      clashingProperty,
-      clashingTimes
+    let index = summaryErrors.findIndex(
+      (error) => error.key === 'bothDatesOverlapping'
     );
-    combinedErrors[inputNames.shiftFinishTime] = {
-      message: '',
+    const error = {
+      key: 'bothDatesOverlapping',
+      inputName: 'shift-start-time',
+      message: clashErrorMessage(
+        clashingProperty,
+        clashingTimes,
+        timePeriodTypesMap
+      ),
+      errorPriority: 1,
     };
+
+    combinedErrors = [
+      ...combinedErrors.slice(0, index),
+      error,
+      ...combinedErrors.slice(index + 1),
+    ];
```

Comment:
This is repeated three times - could it move outside of the if statements?

Reply:
split out into a function